packages/bundle-analyzer/src/assets.ts

```diff
@@ -69,11 +88,38 @@ export const listChildFilePaths = async (
 
     if (file.isDirectory()) {
       const childPaths = await listChildFilePaths(fullPath);
-      childPaths.forEach((file) => results.push(file));
+      childPaths.forEach((childFile) => results.push(childFile));
     } else if (file.isFile()) {
       results.push(fullPath);
     }
   }
 
   return results;
 };
+
+/* getAllAssets consolidates assets from multiple directories (including additionalBuildDirectories) */
+export const getAllAssets = async (
+  buildDirectoryPath: string,
+  additionalBuildDirectories: string[] = [],
+  ignorePatterns: string[] = [],
+  normalizeAssetsPattern = "",
+): Promise<Asset[]> => {
+  // get assets from the main build directory
+  const allAssets = await getAssets(
+    buildDirectoryPath,
+    ignorePatterns,
+    normalizeAssetsPattern,
+  );
+
+  // get assets from any additional build directories
+  for (const additionalDir of additionalBuildDirectories) {
+    const additionalAssets = await getAssets(
+      additionalDir,
+      ignorePatterns,
+      normalizeAssetsPattern,
+    );
+    allAssets.push(...additionalAssets);
+  }
```

Comment:
We should `Promise.all` this block so we can parallelize `getAssets`.

`const allAssets = await Promise.all([getAssets(...), additionalBuildDirectories.map(...)])`

Reply:
fixed!